.github/workflows/self-push.yml

```diff
@@ -7,6 +7,7 @@ on:
     types: [completed]
```

Comment:
I recommend to have `self-push-amd.yml` which is copied from `self-push.yml` but with the modification to use AMD related stuff.

So the Nvidia/AMD jobs won't be mixed in a single file. More advantages:

- The jobs will be clear to navigate (no nvidia/amd jobs being all in the same run page)
- We don't need to rework the Slack report script `utils/notification_service.py` (which is already too complex IMO)
  - we don't need to rename `machine_type`  (and the tags) to `single-amdgpu` / `multi-amdgpu` (this requires rework on ``utils/notification_service.py` to make the report working
  - we only need to tag `docker-amdgpu` and `single-gpu` (or `multi-gpu`) on the runner, and use `machine_type: [ single-gpu ]` (or `machine_type: [ mult-amdgpu ]`)

Reply:
Then both nvidia/amd workflow will be triggered (at the same time) independently, see


[.github/workflows/self-push-caller.yml](https://github.com/huggingface/transformers/blob/main/.github/workflows/self-push-caller.yml)